apps/designer/app/shared/db/assets.server.ts

```diff
@@ -15,11 +18,30 @@ export const loadByProject = async (projectId?: Project["id"]) => {
 
 export const create = async (
   projectId: Project["id"],
-  values: { type: string; name: string; path: string }
+  values: { name: string; path: string },
+  absolutePath?: string
 ) => {
+  let size = 0;
+  const image = await sharp(absolutePath);
+  const metadata = await image.metadata();
+
+  // if there is no absolute path then it means it's a remote image and we can use fetch
+  if (absolutePath) {
+    size = await (await fsPromises.stat(absolutePath)).size;
+  } else {
+    const arrayBuffer = await fetch(values.path).then((rsp) =>
+      rsp.arrayBuffer()
+    );
+    size = arrayBuffer.byteLength;
+  }
+
```

Comment:
I am sure we don't have to do it, its all available during upload, because browser sends that info, just need to figure out how to get that

Reply:
Updated everything

way better now